src/Fixtures/AbstractFixture.php

```diff
@@ -0,0 +1,45 @@
+<?php declare(strict_types=1);
+
+namespace Neusta\Pimcore\FixtureBundle\Fixtures;
+
+use Doctrine\Bundle\FixturesBundle\ORMFixtureInterface;
+use Doctrine\Common\DataFixtures\FixtureInterface;
+use Neusta\Pimcore\FixtureBundle\ReferenceRepository\DataObjectReferenceRepository;
+use Symfony\Contracts\Service\Attribute\Required;
+
+abstract class AbstractFixture implements FixtureInterface, ORMFixtureInterface
+{
+    private DataObjectReferenceRepository $dataObjectReferenceRepository;
+
+    #[Required]
+    public function setDataObjectReferenceRepository(DataObjectReferenceRepository $dataObjectReferenceRepository): void
+    {
+        $this->dataObjectReferenceRepository = $dataObjectReferenceRepository;
+    }
+
+    protected function setReference(string $name, object $reference): void
+    {
+        $this->dataObjectReferenceRepository->setReference($name, $reference);
+    }
+
+    protected function addReference(string $name, object $object): void
+    {
+        $this->dataObjectReferenceRepository->addReference($name, $object);
+    }
+
+    /**
+     * @psalm-param class-string $class
+     */
+    protected function getReference(string $name, string $class)
+    {
+        return $this->dataObjectReferenceRepository->getReference($name, $class);
+    }
+
+    /**
+     * @psalm-param class-string $class
+     */
+    protected function hasReference(string $name, string $class): bool
+    {
+        return $this->dataObjectReferenceRepository->hasReference($name, $class);
+    }
```

Comment:
What happens if you want to reference ORM objects? That's currently not possible, right?

Reply:
That is correct. There is a different Reference-Repository for Doctrine Entities. If we need both Pimcore-Objects and Doctrine-Entities we should create some sort of Proxy-Repository which can handle both. As we currently do not have the need i would advise against it.